src/refda/exec.c

```diff
@@ -452,3 +452,171 @@ void *refda_exec_worker(void *arg)
     CACE_LOG_INFO("Worker stopped");
     return NULL;
 }
+
+/** Execute a time based rule's action that has already been verified
+ * Based on code from refda_exec_exp_execset
+ */
+static int refda_exec_tbr_action(refda_agent_t *agent, const refda_amm_tbr_desc_t *tbr)
+{
+    refda_runctx_ptr_t ctxptr;
+    refda_runctx_ptr_init_new(ctxptr);
+
+    if (refda_runctx_from(refda_runctx_ptr_ref(ctxptr), agent, NULL))
+    {
+        return 2;
+    }
+
+    cace_ari_list_t *targets = &(tbr->action.as_lit.value.as_ac->items);
```

Comment:
The action of a rule should be treated as a single execution target; so an AC action would be handled as a MAC value (there are already functions to do just this).

Reply:
Are you referring to `refda_exec_exp_mac`? Maybe we can simplify by setting up the sequence in `refda_exec_tbr_action` and then calling into the existing MAC function.

Reply:
Simplified code per my previous comment.